src/index.js

```diff
@@ -22,7 +29,12 @@ export default function livereload (options = { watch: '' }) {
     server.watch(resolve(process.cwd(), options.watch))
   }
 
-  closeServerOnTermination(server)
+  // hooking on SIGINT/SIGTERM might cause more harm than good, since only one
+  // plugin / lib can do that, we might cause process.exit before actual cleanup
+  // is all completed -- disabling by default behind an option for now...
```

Comment:
This should probably be added to the [readme](https://github.com/thgh/rollup-plugin-livereload/blob/4311f23045e8c37335e337a88a89a0d477fb63a7/README.md).

Reply:
Absolutely. So I chickened out of this and restored the current signal handling behaviour in my last commit...

I'm not so sure about what should be the proper handling for termination signals actually. Plus my previous change would be breaking in regard to semver, since it disables a behaviour that was previously present by default. And the option should probably be given a better name (`handleTerminationSignal` ?). IMHO this should even be removed entirely, instead of making it an option...

All in all, I think signal handling would deserve more reflexion, and it only fixes a theoretical problem as far as this plugin is concerned. The real problem that people have been reporting is restarting the server when Rollup restarts, so I'll focus this PR on this quick win.

Do you want another PR to remove termination signal handling or make it optional?

Reply:
To be clear I’m not a maintainer of the package! I think the PR looks great and I keep running into the problem it seems to fix. I just noticed that the API change ought to be documented.

Reply:
Oh? Right! You're someone else with a nickname of 't' + random consonants! What are the chances?

Anyway, I think your remark was relevant, and it was a bad idea to mix a small fix that people are awaiting with an unneeded API change.

I'm marking this discussion as resolved (feel free to reopen if I've missed something), and updating the PR's description instead.

Reply:
This PR and 34 change the behaviour so I would make it v2 with or without termination signals.